Update table and Aggregates

Yosefgeda opened the pull request:
- Updates are done for the animals' table in the vet-clinic database.
- Added a column species of type string.

- Made the following updates: 

- Set the species value to unspecified.
![CHANGED SPECIES UNSPECIFIED](https://github.com/Yosefgeda/vet-clinic/assets/100337086/849bd9d5-66e7-4bb7-86e0-cf10157034a5)
- species values after ROLLBACK.
![CHANGED SPECIES UNSPECIFIED after rollback](https://github.com/Yosefgeda/vet-clinic/assets/100337086/1a47b915-5997-4404-bf33-c120f24fc98c)

1. Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
![1](https://github.com/Yosefgeda/vet-clinic/assets/100337086/62dc9c74-ef03-4556-82fd-a5f718b80d5e)

2. Update the animals table by setting the species column to Pokemon for all animals that don't have species already set.
![2](https://github.com/Yosefgeda/vet-clinic/assets/100337086/f8fb5c66-e391-4fe8-ade4-c7aee31195de)

3. Commit the transaction.
![3](https://github.com/Yosefgeda/vet-clinic/assets/100337086/96be8489-05dd-4e05-8f80-5f6ec44482f1)

4. Inside a transaction delete all records in the animals table, then roll back the transaction.
![4](https://github.com/Yosefgeda/vet-clinic/assets/100337086/028db786-7eaa-4918-a838-26b19823b4ac)

5. After the rollback verify if all records in the animals table still exist. After that, you can start breathing as usual ;)
![5](https://github.com/Yosefgeda/vet-clinic/assets/100337086/0f947b5c-9c30-4967-973b-69f5e8a2dadf)

6. Delete all animals born after Jan 1st, 2022.
![6](https://github.com/Yosefgeda/vet-clinic/assets/100337086/5d5cc49e-b709-4323-8b53-2933b0f8ac07)

7. Create a savepoint for the transaction.
![7](https://github.com/Yosefgeda/vet-clinic/assets/100337086/fb77454e-0c8d-4b40-8650-38bb77735c25)

8. Update all animals' weights to be their weight multiplied by -1.
![multiply all by negative](https://github.com/Yosefgeda/vet-clinic/assets/100337086/b9eaf216-e8f2-4fb3-975d-af8562c8428a)

9. How many animals are there?
![number of animals](https://github.com/Yosefgeda/vet-clinic/assets/100337086/63ec9191-d4db-4888-be0e-81a17be12a91)

10. How many animals have never tried to escape?
![no escape attempts](https://github.com/Yosefgeda/vet-clinic/assets/100337086/445c589a-8c36-463d-ad7e-6d65ec66e5e7)

11. What is the average weight of animals?
![CHANGED AVG WEIGHT](https://github.com/Yosefgeda/vet-clinic/assets/100337086/a844ab34-dfa7-44ce-b729-8154a88856e4)

12. Who escapes the most, neutered or not neutered animals?
![who escapes the most](https://github.com/Yosefgeda/vet-clinic/assets/100337086/4d31d578-480f-49a5-8b8f-965c173c565a)

13. What is the minimum and maximum weight of each type of animal?

![CHANGED MAX WEIGHT](https://github.com/Yosefgeda/vet-clinic/assets/100337086/98018cf4-5a13-4ca7-bde5-a544780dd13d)

![CHANGED MIN WEIGHT](https://github.com/Yosefgeda/vet-clinic/assets/100337086/8362a039-6133-4e5f-b988-ae3b28596c7f)

14. What is the average number of escape attempts per animal type of those born between 1990 and 2000?

![avg escpae attempts](https://github.com/Yosefgeda/vet-clinic/assets/100337086/bceea56e-b23b-497c-ac0f-be15799afd9b)

Olamarx reviewed (changes requested):
Hi @Yosefgeda,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!
<img src="https://media0.giphy.com/media/KyBesf8kNT6s88SEgb/giphy.gif"/>

## Highlights
- Query database ✔️
- All added to appropriate file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Yosefgeda commented on queries.sql:
```diff
@@ -7,4 +7,69 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+/*update the animals table by setting the species column to unspecified*/
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
```
![CHANGED SPECIES UNSPECIFIED](https://github.com/Yosefgeda/vet-clinic/assets/100337086/80736394-eeae-4996-bb81-07854cce11b7)
After ROLLBACK
![CHANGED SPECIES UNSPECIFIED after rollback](https://github.com/Yosefgeda/vet-clinic/assets/100337086/28fa987b-2c2a-4f4e-bb66-17c46016e6f8)

Yosefgeda commented on queries.sql:
```diff
@@ -7,4 +7,69 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+Commit transaction 
+*/
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT update_table;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO update_table;
+SELECT * FROM animals;
+COMMIT;
```
![multiply all by negative AND ROLLBACK](https://github.com/Yosefgeda/vet-clinic/assets/100337086/a7bdaad2-dcbc-4a6a-bee6-9d81b536883a)

Yosefgeda commented on queries.sql:
```diff
@@ -7,4 +7,69 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+
+/* What is the average weight of animals? */
+
+SELECT AVG (weight_kg) FROM animals;
```
![CHANGED AVG WEIGHT](https://github.com/Yosefgeda/vet-clinic/assets/100337086/a3edab23-7bde-4eb5-8687-e3b2d76d9604)

Yosefgeda commented:
- Multiply all by negative AND ROLLBACK;
![multiply all by negative AND ROLLBACK](https://github.com/Yosefgeda/vet-clinic/assets/100337086/faabc79a-570e-421a-acdf-503493d19f57)
- Changed species to unspecified after rollback
![CHANGED SPECIES UNSPECIFIED after rollback](https://github.com/Yosefgeda/vet-clinic/assets/100337086/e6819e76-363b-49cb-bf7f-6fdb31dc8783)
- Changed average weight
![CHANGED AVG WEIGHT](https://github.com/Yosefgeda/vet-clinic/assets/100337086/43264f84-d0ac-4427-8f7f-3a7288ed0bbc)
- Changed min weight
![CHANGED MIN WEIGHT](https://github.com/Yosefgeda/vet-clinic/assets/100337086/d932369a-d3a8-4b62-a61a-bc3db5c41c8d)
- Changed max weight
![CHANGED MAX WEIGHT](https://github.com/Yosefgeda/vet-clinic/assets/100337086/58396d67-34c3-4809-9e05-e248e91c0bea)

HAMMAS-SALEEM commented on queries.sql:
```diff
@@ -7,4 +7,69 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+/*update the animals table by setting the species column to unspecified*/
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
```
- [ ] It is great that you have added the screenshot but I guess you are supposed to add all the screenshots in the pull request summary as required in the project requirements 👍 Kindly put the screenshots of this query in the pull request summary to follow the project requirements 😊 as suggested by the previous reviewer 🤔
- [ ] It would be nice as well if you replace all the new screenshots with the old screenshots in the pull request summary because i saw that you have added them but not in the pull request summary 😨

HAMMAS-SALEEM commented on queries.sql:
```diff
@@ -7,4 +7,69 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+/*update the animals table by setting the species column to unspecified*/
```
**[OPTIONAL]**

- [ ] It would be nice if you add your own name logo created by you at the top of the readme file to make your readme attractive and present you as a brand in the software industry 👍 to catch the recruiters' eyes 😊

PinkMoon25 reviewed (approved):
Hi @Yosefgeda , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media3.giphy.com/media/NEvPzZ8bd1V4Y/giphy.gif"/>

## Highlights
Gitflow :heavy_check_mark: 
Changes Implemented :heavy_check_mark:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
